src/main/java/org/folio/edge/api/utils/util/ApiKeyParser.java

```diff
@@ -0,0 +1,52 @@
+package org.folio.edge.api.utils.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Base64;
+import org.folio.edge.api.utils.model.ClientInfo;
+
+public class ApiKeyParser {
+
+  private ApiKeyParser() {
+  }
+
+  public static ClientInfo parseApiKey(String apiKey) throws MalformedApiKeyException {
+    ClientInfo clientInfo;
+
+    try {
+      String decoded = new String(Base64.getUrlDecoder().decode(apiKey.getBytes()));
+      ObjectMapper mapper = new ObjectMapper();
+      clientInfo = mapper.readValue(decoded, ClientInfo.class);
+    } catch (Exception var4) {
+      throw new MalformedApiKeyException("Failed to parse", var4);
```

Comment:
Maybe it's better to add more a detailed exception message with info about api key?